src/test/java/com/commercetools/sync/shoppinglists/helpers/ShoppingListReferenceResolverTest.java

```diff
@@ -0,0 +1,396 @@
+
+        // test and assertion
+        assertThat(referenceResolver.resolveCustomerReference(draftBuilder).toCompletableFuture())
+            .hasFailedWithThrowableThat()
+            .isExactlyInstanceOf(ReferenceResolutionException.class)
+            .hasMessage(format(FAILED_TO_RESOLVE_CUSTOMER_REFERENCE, draftBuilder.getKey(),
+                BLANK_KEY_VALUE_ON_RESOURCE_IDENTIFIER));
+    }
+
+    @Test
+    void resolveCustomerReference_WithNonExistingCustomerKey_ShouldNotResolveCustomerReference() {
+        // preparation
+        when(customerService.fetchCachedCustomerId(anyString()))
+            .thenReturn(CompletableFuture.completedFuture(Optional.empty()));
+
+        final String customerKey = "non-existing-customer-key";
+        final ShoppingListDraftBuilder draftBuilder =
+            ShoppingListDraftBuilder
+                .of(LocalizedString.of(Locale.ENGLISH, "NAME"))
+                .customer(ResourceIdentifier.ofKey(customerKey))
+                .description(LocalizedString.of(Locale.ENGLISH, "DESCRIPTION"));
+
+        // test and assertion
+        assertThat(referenceResolver.resolveCustomerReference(draftBuilder).toCompletableFuture())
+            .hasFailedWithThrowableThat()
+            .isExactlyInstanceOf(ReferenceResolutionException.class)
+            .hasMessage(format(ShoppingListReferenceResolver.FAILED_TO_RESOLVE_CUSTOMER_REFERENCE,
+                draftBuilder.getKey(), format(CUSTOMER_DOES_NOT_EXIST, customerKey)));
+    }
+
+    @Test
+    void resolveCustomerReference_WithIdOnCustomerReference_ShouldNotResolveReference() {
+        // preparation
+        final ShoppingListDraftBuilder draftBuilder =
+            ShoppingListDraftBuilder
+                .of(LocalizedString.of(Locale.ENGLISH, "NAME"))
+                .customer(ResourceIdentifier.ofId("existingId"))
+                .description(LocalizedString.of(Locale.ENGLISH, "DESCRIPTION"));
+
+        // test and assertion
+        assertThat(referenceResolver.resolveCustomerReference(draftBuilder).toCompletableFuture())
+            .hasNotFailed()
+            .isCompletedWithValueMatching(resolvedDraft ->
+                Objects.equals(resolvedDraft.getCustomer(), draftBuilder.getCustomer()));
+    }
+
+    @Test
+    void resolveReferences_WithoutReferences_ShouldNotResolveReferences() {
+        final ShoppingListDraft shoppingListDraft = ShoppingListDraftBuilder
+            .of(LocalizedString.ofEnglish("name"))
+            .key("shoppingList-key")
+            .build();
+
+        final ShoppingListDraft resolvedDraft = referenceResolver
+            .resolveReferences(shoppingListDraft)
+            .toCompletableFuture()
+            .join();
+
+        assertThat(resolvedDraft).isEqualTo(shoppingListDraft);
+    }
+
+    @Test
+    void resolveReferences_WithAllValidFieldsAndReferences_ShouldResolveReferences() {
+        final CustomFieldsDraft customFieldsDraft = CustomFieldsDraft
```

Comment:
// preparation, test and assertions

Reply:
Can you check in general all the test files and add those sections where missing?

Reply:
you are right, I am not a fan of those comments, all of them removed:  https://github.com/commercetools/commercetools-sync-java/pull/602/commits/992d11e4290aaad05188839eff64777e4de88d0a

Reply:
note: to help you, I resolved similar comments. Let's discuss it here.